### PR TITLE
ERC-20: allow admin to change metadata

### PR DESCRIPTION
**Pull request by sept-en**

### CODE REVIEW

File: etc/eth-contracts/contracts/EvmErc20.sol [hunk 1]
```diff
@@ -13,16 +13,37 @@ import "./IExit.sol";
  * `ERC20` functions.
  */
 contract EvmErc20 is ERC20, AdminControlled, IExit {
+    string private _name;
```
Comment (0x3bfc) [inline etc/eth-contracts/contracts/EvmErc20.sol:16]: I think we don't need this here. It is already in [ERC20](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L40)

File: etc/eth-contracts/contracts/EvmErc20.sol [hunk 2]
```diff
@@ -13,16 +13,37 @@ import "./IExit.sol";
...
+    constructor (string memory metadata_name, string memory metadata_symbol, uint8 metadata_decimals, address admin)
+        ERC20(metadata_name, metadata_symbol)
+        AdminControlled(admin, 0)
+    {
+        _name = metadata_name;
+        _symbol = metadata_symbol;
+        _decimals = metadata_decimals;
+    }
+
+    function name() public view override returns (string memory) {
```
Comment (0x3bfc) [inline etc/eth-contracts/contracts/EvmErc20.sol:29]: These functions already exists in the inherited ERC20 contract.

File: etc/eth-contracts/contracts/EvmErc20.sol [hunk 3]
```diff
@@ -13,16 +13,37 @@ import "./IExit.sol";
...
+        return _name;
+    }
+
+    function symbol() public view override returns (string memory) {
```
Comment (0x3bfc) [inline etc/eth-contracts/contracts/EvmErc20.sol:33]: the same here

**Review by 0x3bfc — CHANGES_REQUESTED** [review]
Minor changes

File: etc/eth-contracts/contracts/EvmErc20.sol (hunk 1 above)
Comment (sept-en) [reply etc/eth-contracts/contracts/EvmErc20.sol:16]: This field is `private` there

**Review by 0x3bfc — APPROVED** [review]

File: etc/eth-contracts/contracts/EvmErc20.sol (hunk 2 above)
Comment (sept-en) [reply etc/eth-contracts/contracts/EvmErc20.sol:29]: We need to override the function as the field it referenced is `private` there

File: etc/eth-contracts/contracts/EvmErc20.sol (hunk 3 above)
Comment (sept-en) [reply etc/eth-contracts/contracts/EvmErc20.sol:33]: We need to override the function as the field it referenced is `private` there

File: etc/eth-contracts/hardhat.config.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-require('@nomiclabs/hardhat-ethers');
+require('@nomiclabs/hardhat-waffle');
```
Comment (mfornet) [inline etc/eth-contracts/hardhat.config.js:1]: Out of curiosity, why using waffle here, instead of ethers?

**Review by mfornet — APPROVED** [review]

**Review by birchmd — APPROVED** [review]

File: etc/eth-contracts/hardhat.config.js (hunk 4 above)
Comment (sept-en) [reply etc/eth-contracts/hardhat.config.js:1]: We need to have `hardhat-waffle` as this is adds additional functionality to `chai` (e.g. `.reverted`).
There's no need for `require("@nomiclabs/hardhat-ethers")` as `@nomiclabs/hardhat-waffle` already does it.

**Review by artob — APPROVED** [review]
LGTM :+1:

As a general note, as discussed with @sept-en today, we ought to be checking in the resulting Solidity bytecode as well, since it isn't necessarily 100% reproducible otherwise. I'll add that.

**Pull request merged**